ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -263,16 +255,44 @@ public void startup() throws InstantiationException {
          * having a separate CloudtoStorageManager for cloud offers the flexibility to override methods and add more
          * suited to cloud.
          */
-        networkClientFactory = new RecoveryNetworkClientFactory(properties, registry, clusterMap, storageManager,
-            accountService);
-        vcrClusterSpectator = null; // Server does not talk to vcr during recovery
-        _recoveryManager =
-            new RecoveryManager(replicationConfig, clusterMapConfig, storeConfig, storageManager,
-                storeKeyFactory, clusterMap, scheduler, nodeId, networkClientFactory, registry, notificationSystem,
-                storeKeyConverterFactory, serverConfig.serverMessageTransformer, null,
-                clusterParticipants.get(0));
-        _recoveryManager.start();
-      } else {
+        networkClientFactory =
+            new RecoveryNetworkClientFactory(properties, registry, clusterMap, storageManager, accountService);
+        recoveryManager =
```

Comment:
Recovery manager is not copying from VCR anymore, it copies data directly from cloud. It's time to clean up some RecoveryManager code, we still have some code handles VCR nodes etc..

Reply:
good pt ! removed the vcr-cluster-spectator.  i assume that's what u r referring to.